client/app/hosting/hosting.service.js

```diff
@@ -590,13 +590,12 @@
              * @param {string} serviceName
              * @param {string} offer
              * @param {string} duration
+             * @param {string} startTime (optional)
              */
-      orderUpgrade(serviceName, offer, duration) {
+      orderUpgrade(serviceName, offer, duration, startTime = null) {
         return this.OvhHttp.post(`/order/hosting/web/${serviceName}/upgrade/${duration}`, {
           rootPath: 'apiv6',
-          data: {
-            offer,
-          },
+          data: Object.assign({ offer }, startTime ? { startTime } : {}),
```

Comment:
```suggestion
          data: {
              ...offer
              startTime: startTime ? startTime : {}
          },
```
I like this way better :)

Reply:
I think you mean:
```
data: {
  offer,
  ...(startTime ? { startTime } : {})
},
```

Reply:
Let's find a compromise:
````js
data: {
  offer,
  startTime:  startTime || {},
}
````
😅